src/use_case/refactor_project/RefactorProjectInteractor.java

```diff
@@ -0,0 +1,103 @@
+package use_case.refactor_project;
+import entity.*;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+
+    public class RefactorProjectInteractor implements RefactorProjectInputBoundary {
+        final private RefactorProjectDataAccessInterface userDataAccessObject;
+        final private RefactorProjectOutputBoundary userPresenter;
+        final private ProjectFactory projectFactory;
+        final private TaskFactory taskFactory;
+
+        public RefactorProjectInteractor(RefactorProjectDataAccessInterface refactorProjectDataAccessInterface,
+                                         RefactorProjectOutputBoundary refactorProjectOutputBoundary, ProjectFactory projectFactory, TaskFactory taskFactory) {
+            this.userDataAccessObject = refactorProjectDataAccessInterface;
+            this.userPresenter = refactorProjectOutputBoundary;
+            this.projectFactory=projectFactory;
+            this.taskFactory=taskFactory;
+        }
+
+        @Override
+        public void execute(RefactorProjectInputData refactorProjectInputData) {
+            // Getting the time right now
+            LocalDateTime now = LocalDateTime.now();
+            // getting the user and the project Id from the dao
+            User user = userDataAccessObject.getUser(refactorProjectInputData.getUserName());
+            String projectID = refactorProjectInputData.getId();
+            ArrayList<Project> projects = new ArrayList<>();
+            projects = user.getProjects();
```

Comment:
there is no reason to initialize projects to an empty ArrayList just to set it to user.getProjects() after.